src/requestManager.cpp

```diff
@@ -329,74 +346,78 @@ void RequestBlock(CNode* pfrom, CInv obj)
-		  LogPrint("thin", "Requesting Thinblock %s from peer %s (%d)\n", inv2.hash.ToString(), pfrom->addrName.c_str(),pfrom->id);
-		}
-	    }
-	  else 
-	    {
-	      // Try to download a thinblock if possible otherwise just download a regular block
-	      if (pfrom->mapThinBlocksInFlight.size() < 1 && CanThinBlockBeDownloaded(pfrom)) { // We can only send one thinblock per peer at a time
-		  pfrom->mapThinBlocksInFlight[inv2.hash] = GetTime();
-		  inv2.type = MSG_XTHINBLOCK;
-		  std::vector<uint256> vOrphanHashes;
-                  {
-                    LOCK(cs_orphancache);
-                    for (map<uint256, COrphanTx>::iterator mi = mapOrphanTransactions.begin(); mi != mapOrphanTransactions.end(); ++mi)
-                        vOrphanHashes.push_back((*mi).first);
-                  }
-		  BuildSeededBloomFilter(filterMemPool, vOrphanHashes,inv2.hash);
-		  ss << inv2;
-		  ss << filterMemPool;
-		  pfrom->PushMessage(NetMsgType::GET_XTHIN, ss);
-		  LogPrint("thin", "Requesting Thinblock %s from peer %s (%d)\n", inv2.hash.ToString(), pfrom->addrName.c_str(),pfrom->id);
-	        }
-	      else 
-		{
-		  LogPrint("thin", "Requesting Regular Block %s from peer %s (%d)\n", inv2.hash.ToString(), pfrom->addrName.c_str(),pfrom->id);
-		  std::vector<CInv> vToFetch;
-		  inv2.type = MSG_BLOCK;
-		  vToFetch.push_back(inv2);
-		  pfrom->PushMessage(NetMsgType::GETDATA, vToFetch);
-		}
-	      MarkBlockAsInFlight(pfrom->GetId(), obj.hash, chainParams.GetConsensus());
-	    }
-	}
+  // BUIP010 Xtreme Thinblocks: begin section
+  CInv inv2(obj);
+  CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+  CBloomFilter filterMemPool;
+  if (IsThinBlocksEnabled() && IsChainNearlySyncd()) 
+    {
+      if (HaveConnectThinblockNodes() || (HaveThinblockNodes() && thindata.CheckThinblockTimer(obj.hash))) 
+        {
+          // Must download a block from a ThinBlock peer
+          if (pfrom->mapThinBlocksInFlight.size() < 1 && CanThinBlockBeDownloaded(pfrom)) 
+            { // We can only send one thinblock per peer at a time
+              pfrom->mapThinBlocksInFlight[inv2.hash] = GetTime();
+              inv2.type = MSG_XTHINBLOCK;
+              std::vector<uint256> vOrphanHashes;
+              {
+                LOCK(cs_orphancache);
+                for (map<uint256, COrphanTx>::iterator mi = mapOrphanTransactions.begin(); mi != mapOrphanTransactions.end(); ++mi)
+                    vOrphanHashes.push_back((*mi).first);
+              }
+              BuildSeededBloomFilter(filterMemPool, vOrphanHashes,inv2.hash);
+              ss << inv2;
+              ss << filterMemPool;
+              pfrom->PushMessage(NetMsgType::GET_XTHIN, ss);
+              MarkBlockAsInFlight(pfrom->GetId(), obj.hash, chainParams.GetConsensus());
+              LogPrint("thin", "Requesting Thinblock %s from peer %s (%d)\n", inv2.hash.ToString(), pfrom->addrName.c_str(),pfrom->id);
+              return true;
+            }
+        }
```

Comment:
Congratulation, you broken the whole indentation. Which is bad in itself, but has the added benefit of making the diff completely impossible to review. Please fix.